src/evaluate/module.py

```diff
@@ -436,7 +452,12 @@ def compute(self, *, predictions=None, references=None, **kwargs) -> Optional[di
 
             inputs = {input_name: self.data[input_name] for input_name in self._feature_names()}
             with temp_seed(self.seed):
-                output = self._compute(**inputs, **compute_kwargs)
+                config_state = deepcopy(self.config)
+                self.config.update(compute_kwargs)
```

Comment:
Maybe use a temporary assignment here ? Otherwise calling `compute` twice, first with kwargs and then without, would apply the kwargs to the second call

Reply:
That's why after the call the config is reverted: `self._module_info.config = config_state` in L460. Or do you see a flaw in that logic?

Reply:
If there's an error during `_compute`, then it's not going back to normal, you can use `try:... finally:...`